packages/server/core/src/base/adapters/node/node.ts

```diff
@@ -0,0 +1,154 @@
+import { Server as NodeServer, createServer, ServerResponse } from 'node:http';
+import { NodeRequest, NodeResponse } from '../../../core/plugin';
+import { RequestHandler } from '../../../core/server';
+import {
+  createReadableStreamFromReadable,
+  writeReadableStreamToWritable,
+} from './polyfills/stream';
+import { installGlobals } from './polyfills/install';
+
+installGlobals();
+
+export const createWebRequest = (
+  req: NodeRequest,
+  res: NodeResponse,
+): Request => {
+  const headerRecord: [string, string][] = [];
+  const len = req.rawHeaders.length;
+  for (let i = 0; i < len; i += 2) {
+    headerRecord.push([req.rawHeaders[i], req.rawHeaders[i + 1]]);
+  }
+
+  const { method } = req;
+
+  const controller = new AbortController();
+  const init: RequestInit = {
+    method,
+    headers: headerRecord,
+    signal: controller.signal,
+  };
+  res.on('close', () => controller.abort());
+
+  // TODO: Since we don't want break changes and now node.req.body will be consumed in bff, custom server, render, so we don't create a stream and consume node.req here now.
+  if (
+    !(method === 'GET' || method === 'HEAD') &&
+    req.url?.includes('__loader')
+  ) {
+    init.body = createReadableStreamFromReadable(req);
+    (init as { duplex: 'half' }).duplex = 'half';
+  }
+
+  const url = `http://${req.headers.host}${req.url}`;
```

Comment:
On the server side, don't we need protocol?

Reply:
It seems request by `new Request(url)`, URL must include protocol

Reply:
Oh, i mean use `req.protocol`, not hard code 'http'

Reply:
We can write "http://", because:

1. Our server is an http server
2. Users can't get protocol
3. We can only know whether it's http or https by hack way